src/framework/theme/components/cdk/overlay/dynamic/dynamic-overlay.ts

```diff
@@ -0,0 +1,193 @@
+      this.updateContext();
+    }
+  }
+
+  setContentAndContext(content: NbOverlayContent, context: Object) {
+    this.content = content;
+    this.context = context;
+    if (this.container) {
+      this.updateContext();
+    }
+  }
+
+  setComponent(componentType: Type<NbRenderableContainer>) {
+    this.componentType = componentType;
+
+    // in case the component is shown we recreate it and show it back
+    if (this.ref && this.isAttached) {
+      this.dispose();
+      this.show();
+    } else if (this.ref && !this.isAttached) {
+      this.dispose();
+    }
+  }
+
+  setPositionStrategy(positionStrategy: NbAdjustableConnectedPositionStrategy) {
+    this.positionStrategyChange$.next();
+
+    this.positionStrategy = positionStrategy;
+
+    this.positionStrategy.positionChange
+      .pipe(
+        takeWhile(() => this.alive),
+        takeUntil(this.positionStrategyChange$),
+        filter(() => !!this.container),
+      )
+      .subscribe((position: NbPosition) => patch(this.container, { position }));
+
+    if (this.ref) {
+      this.ref.updatePositionStrategy(this.positionStrategy);
+    }
+  }
+
+  show() {
+    if (!this.ref) {
+      this.createOverlay();
+    }
+
+    this.renderContainer();
+  }
+
+  hide() {
+    if (!this.ref) {
+      return;
+    }
+
+    this.ref.detach();
+    this.container = null;
+  }
+
+  toggle() {
+    if (this.isAttached) {
+      this.hide();
+    } else {
+      this.show();
+    }
+  }
+
+  dispose() {
+    this.alive = false;
+    this.hide();
+    if (this.ref) {
+      this.ref.dispose();
+      this.ref = null;
+    }
+  }
+
+  getContainer() {
+    return this.container;
+  }
+
+  protected createOverlay() {
+    this.ref = this.overlay.create({
+      positionStrategy: this.positionStrategy,
+      scrollStrategy: this.overlay.scrollStrategies.reposition(),
+    });
+  }
+
+  protected renderContainer() {
+    const containerContext = this.createContainerContext();
+    if (!this.container) {
+      this.container = createContainer(this.ref, this.componentType, containerContext, this.componentFactoryResolver);
+    }
+    this.container.instance.renderContent();
+
+    this.updatePositionWhenStable();
+  }
+
+  protected updateContext() {
+    const containerContext = this.createContainerContext();
+    Object.assign(this.container.instance, containerContext);
+    this.container.instance.renderContent();
+    this.container.changeDetectorRef.detectChanges();
+
+    /**
+     * Dimensions of the container may be changed after updating the content, so, we have to update
+     * container position. But we need to delay the execution because of how the portalOverlay gets detached
+     * */
+    this.updatePositionWhenStable();
+  }
+
+  protected createContainerContext(): Object {
+    return {
+      content: this.content,
+      context: this.context,
+      cfr: this.componentFactoryResolver,
+    };
+  }
+
+  protected updatePositionWhenStable() {
+    this.zone.onStable
```

Comment:
![image](https://user-images.githubusercontent.com/17410089/51590346-a2976980-1efa-11e9-9ad4-a2da3a6a250a.png)
And it still doesn't work properly.